controllers/controllers.go

```diff
@@ -18,10 +18,45 @@ package controllers
 
 import (
 	"errors"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	"github.com/openshift-kni/numaresources-operator/pkg/status"
 )
 
+type reconcileStep struct {
+	Done          bool // is reconciliation process done after this step?
```

Comment:
do we really need this considering the condition type says it all and there is 1:1 relation for when done is false (condition is available)?

Reply:
same for the error